java/code/src/com/suse/manager/webui/controllers/AnsibleController.java

```diff
@@ -345,6 +372,20 @@ public static String introspectInventory(Request req, Response res, User user) {
         }
     }
 
+    /**
+     * Parse Ansible Inventory content to look for host names
+     *
+     * @param inventoryMap the Ansible Inventory content
+     * @return the Set of hostnames
+     */
+    public static Set<String> parseInventoryAndGetHostnames(Map<String, Map<String, Object>> inventoryMap) {
+        // Assumption: "_meta" and the nested "hostvars" keys always present in the map to contains all hostnames
+        if (inventoryMap.keySet().contains("_meta") && inventoryMap.get("_meta").keySet().contains("hostvars")) {
```

Comment:
- Nit: you can use `containsKey` method instead of getting the `keySet`.
- I think it'd be also worth checking if the nested thing is a map, just to be more sure, that the casting doesn't blow up.

```suggestion
        if (inventoryMap.containsKey("_meta") && inventoryMap.get("_meta").containsKey("hostvars") &&
                inventoryMap.get("_meta").get("hostvars") instanceof Map) {
```

Reply:
It makes definitely sense, thank you.